gateway/ln-gateway/proto/gateway_lnrpc.proto

```diff
@@ -279,6 +281,23 @@ message GetLnOnchainAddressResponse {
   string address = 1;
 }
 
+message WithdrawOnchainRequest {
+  // The address to withdraw funds to.
+  string address = 1;
+
+  // The amount to withdraw in sats.
+  // If 0/empty, the entire balance will be swept.
+  optional uint64 amount_sats = 2;
```

Comment:
Nit: you can use protobuf's `oneof` which allows you to specify an enum. Might make a little more sense because you wouldn't need to use a magic value here. In the generated rust code, it still uses a `Option` though

Reply:
Are you thinking of something like this?

```
message WithdrawOnchainRequest {
  // The address to withdraw funds to.
  string address = 1;

  // The amount to withdraw. Either `amount_sats` or `all` must be set.
  oneof amount {
    // The amount to withdraw in sats.
    uint64 amount_sats = 2;

    // Withdraw all funds.
    bool all = 3;
  }

  // The fee rate to use for the withdrawal transaction.
  uint64 fee_rate_sats_per_vbyte = 4;
}
```

This produces the following type:

```
    pub enum Amount {
        /// The amount to withdraw in sats.
        #[prost(uint64, tag = "2")]
        AmountSats(u64),
        /// Withdraw all funds.
        #[prost(bool, tag = "3")]
        All(bool),
    }
```

I don't like this as much, since the rust type changes from an `Option<u64>` to an `Option<Amount>`. The latter type has two invalid states (`None` and `Some(Amount::All(false))`) whereas the former has none. I lean towards keeping it as an `optional uint64`. WDYT?

Reply:
That is fine. I don't really like magic values but I agree `oneof` is a bit cumbersome. Another reason to get rid of grpc and switch to axum.